### DISCUSSION TITLE
Client quit with exit code 2 and signal 0

### DISCUSSION THREAD

**Discussion by Dante983** [LazyVim/LazyVim#2014] [Q&A] [answered]
Hey everyone, as you can see from the title, i get these weird warnings not sure what they do, and lazy seems kinda slow, so i'm wondering if i'm doing something wrong here?

i've attached a video, and this goes on and on..

https://github.com/LazyVim/LazyVim/assets/51093125/511ee50c-4fae-4ace-a71d-3adb31c0569a

This is my laptop
<img width="262" alt="image" src="https://github.com/LazyVim/LazyVim/assets/51093125/812eaad4-8e08-4f79-9007-e6cfe37874fe">

Let me know if any more info is needed

Also, i 'm wandering if my cursor is acting weird, it's flickering way too much, all the time.

https://github.com/LazyVim/LazyVim/assets/51093125/143bcc6c-1591-414a-a7e0-27c7160d98aa

**Comment by lvl4ul2i** [comment]
I had this problem once, with the :LspInfo command you can see the ids of the clients in that buffer. In my case it was a problem with ruby-ls, a bad installation gave me that exit code.

**Reply by Dante983** [reply]
Not sure, I only have phpactor and easy-coding-standard installed so far, besides lua_ls.
It's a fresh install, so really not sure..

**Comment by dpetka2001** [comment]
Did you copy over your old configuration of Neovim to LazyVim or did you start with a fresh LazyVim starter template? The first error seems to be coming from one of your LSP clients. Plz provide more information about your config. `LspInfo`, as already mentioned, is a good start about what clients are trying to attach to the specific buffer. Regarding your second issue, do you happen to have somewhere in your configuration anything about `guicursor`? If yes, try to disable it and observe behavior.

**Reply by Dante983** [reply]
Here's the LspInfo command
<img width="768" alt="image" src="https://github.com/LazyVim/LazyVim/assets/51093125/8c1ff1b5-6500-4d54-a07e-bc673fb651d5">
Also I have fresh install of LazyVim, i've just added phpactor and easy-coding-standard through Mason, and php, javascript through TSInstall.

Regarding cursor issue, I haven't changed anything in config

**Reply by dpetka2001** [reply]
Do you see any errors in `~/.local/state/nvim/lsp.log`?

**Reply by dpetka2001** [reply]
Also about your cursor blinking maybe it has to do with your terminal? In kitty, for example, there's an option for cursor blinking and if you set that then it will also blink inside Neovim.

**Reply by Dante983** [reply]
[ERROR][2023-11-14 11:32:59] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 11:33:00] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 11:33:01] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[WARN][2023-11-14 11:33:01] .../lua/vim/lsp.lua:1636    "buf_attach_client called on unloaded buffer (id: 10): "
[START][2023-11-14 12:02:59] LSP logging initiated
[ERROR][2023-11-14 12:02:59] .../vim/lsp/rpc.lua:734    "rpc"   "/Users/nikolasavic/.local/share/nvim/mason/bin/phpactor"   "stderr"    "Starting language server, use -vvv for verbose output\n"
[ERROR][2023-11-14 12:02:59] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:02:59] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:00] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:01] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:02] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:04] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:04] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[ERROR][2023-11-14 12:03:04] .../vim/lsp/rpc.lua:734    "rpc"   "node"  "stderr"    "Node.js 18.x is required to run GitHub Copilot but found 16.13.0\n"
[WARN][2023-11-14 12:03:22] .../lua/vim/lsp.lua:1636    "buf_attach_client called on unloaded buffer (id: 10): "
[WARN][2023-11-14 12:03:22] .../lua/vim/lsp.lua:1636    "buf_attach_client called on unloaded buffer (id: 10): "
[WARN][2023-11-14 12:03:22] .../lua/vim/lsp.lua:1636    "buf_attach_client called on unloaded buffer (id: 10): "
[WARN][2023-11-14 12:03:22] .../lua/vim/lsp.lua:1636    "buf_attach_client called on unloaded buffer (id: 10): "

There were much more these errors regarding copilot that seems to want to switch to node v20+.

The issue was copilot plugin, it wants newer version of node, i have v16.* that is currently active.

Thanks guys, log helped me find the issue. 

Also, the cursor is fixed now with it.

**Reply by dpetka2001** [reply, marked as answer]
`Node.js 18.x is required to run GitHub Copilot but found 16.13.0` it states that it needs `node 18.x` not `20.x`. So with `node 18.x` you should be ok.

**Reply by Dante983** [reply]
working on it as we speak :D
